ara/views/host.py

```diff
@@ -21,6 +23,31 @@ def show_host(host):
     except models.NoResultFound:
         abort(404)
 
+    try:
+        facts = host.facts.one()
+    except models.NoResultFound:
+        facts = None
+
     stats = utils.get_host_playbook_stats(host)
 
-    return render_template('host.html', host=host, stats=stats)
+    return render_template('host.html', host=host, stats=stats, facts=facts)
+
+
+@host.route('/<host>/facts')
+def show_facts(host):
+    try:
+        host = models.Host.query.filter_by(name=host).one()
+    except models.NoResultFound:
+        abort(404)
+
+    try:
```

Comment:
Pretty much the same thing here.  All this becomes unnecessary, and you can simply do:

    if not host.facts:
        abort(404)